File: portal/parse.js
const fs = require('fs');

try {
    const raw = fs.readFileSync('.scratch/report2.json', 'utf8');
    const data = JSON.parse(raw);

    data.errors.forEach(e => {
        console.log("================ ERROR ================");
        console.log(e.message || JSON.stringify(e));
        console.log("\n");
    });
} catch (e) {
    console.error("Failed to parse", e);
}

File: portal/parse.py
import json
import codecs

try:
    with open('.scratch/report2.json', 'r', encoding='utf-8') as f:
        # Check for BOM and skip if present
        content = f.read()
        if content.startswith(codecs.BOM_UTF8.decode('utf-8')):
            content = content[1:]
        data = json.loads(content)
        
    for suit in data.get('errors', []):
        print("====== ERROR ======")
        print(suit.get('message', 'No message'))
        
except Exception as e:
    print(f"Failed: {e}")
